File: fix_pickers.py
import re

def main():
    file_path = 'c:/Users/user/Desktop/eduKN/edukn/edukn21.11.2025/edukn/lib/screens/announcements/create_announcement_screen.dart'
    
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Pattern 1
    pattern1 = re.compile(r'final picked = await showDatePicker\(\s*context:\s*context,\s*initialDate:\s*_publishDate,\s*firstDate:[^\)]+\);', re.MULTILINE)
    content = pattern1.sub('final picked = await CustomDateRangePicker.showSingle(context, initialDate: _publishDate);', content)

    # Pattern 2
    pattern2 = re.compile(r'final date = await showDatePicker\(\s*context:\s*context,\s*firstDate:[^\)]+initialDate:[^\)]+\),\s*\);', re.MULTILINE)
    content = pattern2.sub('final date = await CustomDateRangePicker.showSingle(context, initialDate: DateTime.now().add(const Duration(days: 1)));', content)
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
        
    print("Done")

if __name__ == "__main__":
    main()

File: fix_pickers.dart
import 'dart:io';

void main() {
  final file = File('lib/screens/announcements/create_announcement_screen.dart');
  String content = file.readAsStringSync();
  
  // Replace Pattern 1
  final pattern1 = RegExp(r'final picked = await showDatePicker\([\s\S]*?initialDate:\s*_publishDate,[\s\S]*?\);', multiLine: true);
  content = content.replaceFirst(pattern1, 'final picked = await CustomDateRangePicker.showSingle(context, initialDate: _publishDate);');

  // Replace Pattern 2
  final pattern2 = RegExp(r'final date = await showDatePicker\([\s\S]*?initialDate:\s*DateTime\.now\(\)\.add\([\s\S]*?\),[\s\S]*?\);', multiLine: true);
  content = content.replaceFirst(pattern2, 'final date = await CustomDateRangePicker.showSingle(context, initialDate: DateTime.now().add(const Duration(days: 1)));');

  file.writeAsStringSync(content);
  print("Updated create_announcement_screen.dart via regex");
}
